presubmit testing failure: TempDir RemoveAll cleanup: directory not empty (?)

The latest master commit has failed the tests:

```
2024-06-24T11:25:49.1730660Z --- FAIL: TestFuzz (13.86s)
2024-06-24T11:25:49.1731305Z     testutil.go:33: seed=0
2024-06-24T11:25:49.1776279Z ##[error]seed=0
2024-06-24T11:25:49.1789561Z     fuzzer_test.go:71: new prog:
2024-06-24T11:25:49.1792173Z ##[error]new prog:
2024-06-24T11:25:49.1794067Z         syz_test_fuzzer1(0x0, 0x0, 0x0)
2024-06-24T11:25:49.1795170Z     fuzzer_test.go:274: <iter 100>: corpus 1, signal 13, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1797891Z ##[error]<iter 100>: corpus 1, signal 13, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1799802Z     fuzzer_test.go:71: new prog:
2024-06-24T11:25:49.1801712Z ##[error]new prog:
2024-06-24T11:25:49.1803113Z         syz_test_fuzzer1(0x1, 0x1, 0x0)
2024-06-24T11:25:49.1804351Z     fuzzer_test.go:274: <iter 200>: corpus 2, signal 18, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1806999Z ##[error]<iter 200>: corpus 2, signal 18, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1808768Z     fuzzer_test.go:71: new prog:
2024-06-24T11:25:49.1810628Z ##[error]new prog:
2024-06-24T11:25:49.1812737Z         syz_test_fuzzer1(0x1, 0x2, 0x0)
2024-06-24T11:25:49.1813813Z     fuzzer_test.go:274: <iter 300>: corpus 3, signal 20, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1816664Z ##[error]<iter 300>: corpus 3, signal 20, max signal 20, crash types 0, running jobs 4
2024-06-24T11:25:49.1818599Z     fuzzer_test.go:71: new prog:
2024-06-24T11:25:49.1820356Z ##[error]new prog:
2024-06-24T11:25:49.1821921Z         syz_test_fuzzer1(0x1, 0x0, 0x0)
2024-06-24T11:25:49.1822572Z     fuzzer_test.go:265: CRASH: second bug
2024-06-24T11:25:49.1824323Z ##[error]CRASH: second bug
2024-06-24T11:25:49.1826669Z     fuzzer_test.go:274: <iter 400>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 4
2024-06-24T11:25:49.1829254Z ##[error]<iter 400>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 4
2024-06-24T11:25:49.1831495Z     fuzzer_test.go:265: CRASH: second bug
2024-06-24T11:25:52.3766176Z ##[error]CRASH: second bug
2024-06-24T11:25:52.3768926Z     fuzzer_test.go:274: <iter 500>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 4
2024-06-24T11:25:52.3771944Z ##[error]<iter 500>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 4
2024-06-24T11:25:52.3774057Z     fuzzer_test.go:265: CRASH: second bug
2024-06-24T11:25:52.3775939Z ##[error]CRASH: second bug
2024-06-24T11:25:52.3777804Z     fuzzer_test.go:274: <iter 600>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3780683Z ##[error]<iter 600>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3783034Z     fuzzer_test.go:274: <iter 700>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3785803Z ##[error]<iter 700>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3788101Z     fuzzer_test.go:274: <iter 800>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3791603Z ##[error]<iter 800>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3794332Z     fuzzer_test.go:274: <iter 900>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3797174Z ##[error]<iter 900>: corpus 4, signal 20, max signal 20, crash types 1, running jobs 3
2024-06-24T11:25:52.3799085Z     fuzzer_test.go:265: CRASH: first bug
2024-06-24T11:25:52.3800920Z ##[error]CRASH: first bug
2024-06-24T11:25:52.3802542Z     fuzzer_test.go:89: resulting corpus:
2024-06-24T11:25:52.3804266Z ##[error]resulting corpus:
2024-06-24T11:25:52.3806045Z     fuzzer_test.go:91: -----
2024-06-24T11:25:52.3807843Z ##[error]-----
2024-06-24T11:25:52.3854404Z     fuzzer_test.go:92: syz_test_fuzzer1(0x0, 0x0, 0x0)
2024-06-24T11:25:52.3856140Z ##[error]syz_test_fuzzer1(0x0, 0x0, 0x0)
2024-06-24T11:25:52.3857759Z     fuzzer_test.go:91: -----
2024-06-24T11:25:52.3858998Z ##[error]-----
2024-06-24T11:25:52.3860408Z     fuzzer_test.go:92: syz_test_fuzzer1(0x1, 0x1, 0x0)
2024-06-24T11:25:52.3861834Z ##[error]syz_test_fuzzer1(0x1, 0x1, 0x0)
2024-06-24T11:25:52.3863206Z     fuzzer_test.go:91: -----
2024-06-24T11:25:52.3864394Z ##[error]-----
2024-06-24T11:25:52.3865601Z     fuzzer_test.go:92: syz_test_fuzzer1(0x1, 0x2, 0x0)
2024-06-24T11:25:52.3867070Z ##[error]syz_test_fuzzer1(0x1, 0x2, 0x0)
2024-06-24T11:25:52.3868410Z     fuzzer_test.go:91: -----
2024-06-24T11:25:52.3869617Z ##[error]-----
2024-06-24T11:25:52.3870978Z     fuzzer_test.go:92: syz_test_fuzzer1(0x1, 0x0, 0x0)
2024-06-24T11:25:52.3872390Z ##[error]syz_test_fuzzer1(0x1, 0x0, 0x0)
2024-06-24T11:25:52.3874129Z     testing.go:1225: TempDir RemoveAll cleanup: unlinkat /tmp/TestFuzz3649139481/001: directory not empty
2024-06-24T11:25:52.3876616Z ##[error]TempDir RemoveAll cleanup: unlinkat /tmp/TestFuzz3649139481/001: directory not empty
2024-06-24T11:25:52.3878269Z FAIL
2024-06-24T11:25:52.3878593Z exit status 1
2024-06-24T11:25:52.3879074Z FAIL	github.com/google/syzkaller/pkg/fuzzer	15.787s
```

Again I see it on a PR with some absolutely unrelated changes

```
-- FAIL: TestCover (0.21s)
    --- FAIL: TestCover/64 (26.01s)
        testing.go:1225: TempDir RemoveAll cleanup: unlinkat /tmp/TestCover64179800772/001: directory not empty
Error: TempDir RemoveAll cleanup: unlinkat /tmp/TestCover64179800772/001: directory not empty
```

@tarasmadan does it ring any bells with you? Seems like some problem in our presubmit test running infra.

--- a/executor/executor_test.h
+++ b/executor/executor_test.h
@@ -5,11 +5,22 @@
 #include <sys/mman.h>
 #include <unistd.h>
 
+#ifdef __linux__
+#include <sys/prctl.h>
+#endif
+
 static uint64 kernel_text_start = 0xc0dec0dec0000000;
 static uint64 kernel_text_mask = 0xffffff;
 
 static void os_init(int argc, char** argv, void* data, size_t data_size)
 {
+#ifdef __linux__
+	prctl(PR_SET_PDEATHSIG, SIGKILL, 0, 0, 0);
+	// There's a risk that the parent has exited before we get to call prctl().
+	// In that case, let's assume that the child must have been reassigned to PID=1.
+	if (getppid() == 1)
+		exitf("the parent process was killed");
+#endif
 	void* got = mmap(data, data_size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE | MAP_FIXED, -1, 0);
 	if (data != got)
 		failmsg("mmap of data segment failed", "want %p, got %p", data, got);

